src/controllers/export.controller.ts

```diff
@@ -0,0 +1,21 @@
+import { Controller, Get, UseGuards, Res, Header } from '@nestjs/common';
+import { Response } from 'express';
+import { ExportService } from '../services/export.service';
+import { AuthGuard } from '../auth/auth.guard';
+
+@Controller('excel')
+export class ExportController {
+  constructor(private readonly exportService: ExportService) {}
+
+  @UseGuards(AuthGuard)
+  @Get()
+  @Header(
+    'Content-Type',
+    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+  )
+  @Header('Content-Disposition', 'attachment; filename="exampleWorkbook.xlsx"')
```

Comment:
file name is going to be dynamic based on testcase. I'm not sure if you can customize with annotation. if not, you could move this inside the method and set the response header manually. that way you can generate filename dynamically e.g.

```
response.setHeader(
     'Content-Disposition',
     'attachment; filename=filename.xlsx',
   );
  
```

Reply:
Yes, I do have code for making it dynamic but In my ticket I didn't know if I have to do that:

3. @Get(':filename')
4.   @Header(
5.     'Content-Type',
6.     'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
7.   )
8.   @Header('Content-Disposition', 'attachment; filename*=UTF-8\'\'{filename}.xlsx')
9.   async getFile(@Res() res: Response) {

Reply:
gotcha, thanks!